src/wbase/Task.cc

```diff
@@ -454,13 +460,12 @@ bool Task::checkCancelled() {
     return _cancelled;
 }
 
-/// @return true if task has already been cancelled.
-bool Task::setTaskQueryRunner(TaskQueryRunner::Ptr const& taskQueryRunner) {
+bool Task::setTaskQueryRunner(wdb::QueryRunner::Ptr const& taskQueryRunner) {
     _taskQueryRunner = taskQueryRunner;
     return checkCancelled();
 }
 
-void Task::freeTaskQueryRunner(TaskQueryRunner* tqr) {
+void Task::freeTaskQueryRunner(wdb::QueryRunner* tqr) {
     if (_taskQueryRunner.get() == tqr) {
```

Comment:
The row pointer-based comparison (as well as any uses of the C-style row pointers in C++) makes me feel uneasy 😄 I always find it safer to compare objects by their identities, like:
```
if (objA.id() == objB.id()) {
   ...
```

Reply:
I'll add a comment. The pointer is just there to verify the correct thing is being deleted. If `tqr` doesn't match the pointer inside `_taskQueryRunner` nothing is deleted.
It's using the naked pointer as it's being stored inside a lambda function, and using a shared pointer there could cause lifetime issues.